### ISSUE TITLE
We should issue warning on class names that differ only in case

### ISSUE THREAD

**Issue by odersky** [lampepfl/dotty#1077]
Given this program

```
class A
class a

object Test {
  def main(args: Array[String]): Unit = {
    new a
    new A
  }
}
```

scalac issues this warning:

```
pos.scala:2: warning: Class a differs only in case from A. Such classes will overwrite one another on case-insensitive filesystems.
class a
      ^
```

I think dotc should do the same. Also affects `pending/run/t920.scala` which has a similar problem involving class/object on MacOS and Windows.

**Comment by smarter** [comment]
GenBCode has commented-out code to handle this: https://github.com/lampepfl/dotty/blob/3004af37018d2f64c08f33500becf939141ec2e8/compiler/src/dotty/tools/backend/jvm/GenBCode.scala#L190-L202

**Comment by nicolasstucki** [comment]
Fixed by #3897
